tools/packaging/guest-image/build_image.sh

```diff
@@ -32,7 +35,17 @@ build_initrd() {
 	export USE_DOCKER=1
 	export AGENT_INIT="yes"
 	# ROOTFS_BUILD_DEST is a Make variable
-	sudo -E PATH="$PATH" make rootfs ROOTFS_BUILD_DEST="${rootfs_build_dest}"
+
+	if [ -z "${AA_KBC}" == "offline_sev_kbc" ]; then
```

Comment:
@Alex-Carter01 most (if not all) script we set `set -o nounset`. Here if I run the script without exporting `AA_KBC` then it will fail. You must initialize the variable:

`if [ -z "${AA_KBC:-}" == "offline_sev_kbc" ]; then`

Reply:
Actually I've a question about this change... the `build_image.sh` is more like a wrapper for `make rootfs` which in the end is a call to `tools/osbuilder/rootfs-builder/rootfs.sh`. Is there any reason to introduce this platform-specific code here and not in `tools/osbuilder/rootfs-builder/rootfs.sh` (or somewhere on `tools/osbuilder`)?

Here you could export AA_KBC on the make call `sudo -E PATH="$PATH" make rootfs ROOTFS_BUILD_DEST="${rootfs_build_dest} ${AA_KBC:+AA_KBC=$AA_KBC}"`

ps: `${AA_KBC:+AA_KBC=$AA_KBC}` is a little bit crazy, there must be a better way.

Reply:
This is definitely a good discussion to have, so I want to follow up here before making the tweaks to build_image.

I had originally gone back and further between having the logic for locating the efi_secret module in one of three places: kata-deploy-binaries.sh, build_image.sh, and the rootfs.sh.

My thinking had been to keep any platform specific logic out of the rootfs.sh script, and use its KERNEL_MODULES_DIR parameter. Currently the only change I am making to the rootfs.sh script for sev is generic.

The tricky piece of logic in getting the location of the kernel module is using getting the kernel config version. It ends up using the functions which were previously in the kernel builder, but are ultimately reading the kernel config version file. I viewed this as analogous to reading the broader kata version file, which is already being done in the build_image script e.g. to get the distro and os_version.

I am definitely willing to move things around if this belongs better elsewhere.